docs/api/implementation-guide.md

```diff
@@ -0,0 +1,100 @@
+---
+title: NuGet Server Implementation Guide
+description: Guidelines and recommendations to anyone implementing the NuGet Server API in their own package repository
+author: zivkan
+ms.author: zivkan
+ms.date: 07/29/2023
+ms.topic: conceptual
+---
+
+# NuGet Server Implementation Guide
+
+Prior to 2018, NuGet package repositories could copy the package metadata from the `nupkg`'s [`nuspec` file](../reference/nuspec.md), and return the results unmodified in [search](./search-query-service-resource.md) and [package metadata](./registration-base-url-resource.md) requests.
+Since then, NuGet has evolved to provide developers with a richer experience, and this requires package repositories to do additional work in order to provide the additional value to developers.
+
+## Chronology
+
+To assist authors of existing NuGet repositories keep up to date with NuGet's newest features, here is the chronology of the relevant features mentioned in the remainder of the document.
+
+|Version|Feature|
```

Comment:
Package types (returning them in the JSON and filtering by them) were added to search in 2019. Client does not use this but it was around 5.5.0 timeframe. .NET CLI does it it.

ID prefix reservation was implemented in mid to late 2017 and the "verified" boolean was added. 

SemVer 2.0.0 filtering was added in early 2017.

Not sure if you wanted those mentioned.

Reply:
This is utterly perfect, and exactly why I wanted as many people as possible to review.

> SemVer 2.0.0 filtering was added in early 2017.

I wasn't familiar with this before. While all my updates should be reviewed, please pay close attention to this one, because I largely made up everything I wrote based on my best guess and searching the existing resources docs for semver.

Reply:
@joelverhagen should we also add symbols packages info?

This doc has taken me so long, I'm losing motivation. Can I ask someone from the server team to add a section on symbols after I merge this? Especially given client's knowledge of symbols is basically limited to pdbs go in a different nupkg.

Reply:
Probably, yes. We need to find some good links for the symserv HTTP protocol since the NuGet protocol is just ingress. Feel free to open a docs bug and put it on my plate.